core/shared/src/main/scala/zio/ZLogger.scala

```diff
@@ -161,20 +161,17 @@ object ZLogger {
       }
     }
 
-    trace match {
-      case Trace(location, file, line) =>
-        sb.append(" location=")
-
-        appendQuoted(location, sb)
-
-        sb.append(" file=")
-
-        appendQuoted(file, sb)
-
-        sb.append(" line=")
-          .append(line)
-
-      case _ =>
+    val parsedTarce = Trace.parseOrNull(trace)
+    if (parsedTarce ne null) {
+      val location = parsedTarce._1
+      val file     = parsedTarce._2
+      val line     = parsedTarce._3
+
+      sb.append(" location=")
```

Comment:
Do you think it might be possible to pre-size the `StringBuilder` to avoid having it dynamically resized? AFAIK the default initial size is 16 (or maybe 32) which is wayy too small to fit everything